lib/bolt/inventory.rb

```diff
@@ -142,5 +172,22 @@ def expand_targets(targets)
       end
     end
     private :expand_targets
+
+    def init_vars(data)
+      # Replace with &. once we move off ruby 2.0
+      # data['groups']&.each do |group|
+      # group['nodes']&.each do |node|
+      if data['groups'] && data['nodes']
+        data['groups'].each do |group|
+          next unless group['vars']
+          data['nodes'].each do |node|
+            target = get_targets(node)[0]
+            # Instantiate empty vars hash in case no vars are defined
+            @target_vars[target.host] = @target_vars[target.host] || {}
+            group['vars'].map { |k, v| set_var(target, k, v) }
```

Comment:
This design would make "copy on write" inefficient since it means that each `set_var` call would create a new clone. Could be remedied by using a merge, i.e. `@target_vars[target.host] = @target_vars[target.host].merge(group['vars']).freeze`